index.js

```diff
@@ -218,20 +218,26 @@ Client.prototype._pollConfig = function () {
     }
 
     streamToBuffer(res, (err, buf) => {
-      if (err) return res.destroy(err)
+      if (err) {
+        this.emit('request-error', processConfigErrorResponse(res, buf, err))
+        return
```

Comment:
> This means that the agent's handling of this error will go from here, to here

Actually the `res.destroy(err)` would result in an error event on the *socket*, which isn't the Client (the Writable). That error event on the socket would *sometimes* result in an error event on the request object here: https://github.com/elastic/apm-nodejs-http-client/blob/01c621696b35b83d0c692563708bf4e2a88f9193/index.js#L239-L242 which results in the same "request-error" event being emitted.  I say "*sometimes*" because not for the crash case here -- and I certainly having traced possible other cases with/without an agent across 4 versions of node.